### PR TITLE
Add retry support for serializable transactions

### PR DESCRIPTION
**Pull request by neekolas**
## tl;dr

I noticed our `libxmtp` test suite was occasionally failing due to serializable transaction errors. This adds retries, which are required when using the serializable isolation level since errors are common and transient thanks to Postgres' conflict detection

### CODE REVIEW

**Comment by neekolas** [comment]
* **#386** <a href="https://app.graphite.dev/github/pr/xmtp/xmtp-node-go/386?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @neekolas and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: pkg/mls/store/store.go [hunk 1]
```diff
@@ -605,3 +605,20 @@ func (s *Store) RunInTx(
 	done = true
 	return tx.Commit()
 }
+
+func (s *Store) RunInSerializableTx(ctx context.Context, numRetries int, fn func(ctx context.Context, txQueries *queries.Queries) error) error {
+	var err error
+	for i := 0; i < numRetries; i++ {
+		select {
+		case <-ctx.Done():
```
Comment (insipx) [inline pkg/mls/store/store.go:613]: noob Q: What does this select on this ctx do?

The default case returns nil when it finishes, but ctx seems like some sort of global context? When would it trigger?

**Review by insipx — APPROVED** [review]

**Review by 37ng — APPROVED** [review]

File: pkg/mls/store/store.go (hunk 1 above)
Comment (37ng) [reply pkg/mls/store/store.go:613]: If you `ctrl+c` then `ctx.Done()` channel will receive a struct and this `select` block will abort.

Comment (neekolas) [reply pkg/mls/store/store.go:613]: Exactly. It just stops you from continuing retries when the server is in the process of shutting down

**Comment by neekolas** [comment]
The specific tests that were failing are a little worrying to me because they all used random wallets and inboxes, and executed each test serially, so they should be isolated from one another.

We may have to do some debugging around exactly what Postgres considers a "conflict" in the specific queries we are using here. Maybe it's a broader set of rows than you'd expect.

But these retries are a generally good idea regardless and we can tackle that separately.

**Pull request merged**